crypto/fipsmodule/ml_kem/ml_kem.c

```diff
@@ -34,16 +34,36 @@ int ml_kem_512_keypair_deterministic_no_self_test(uint8_t *public_key  /* OUT */
                                                   uint8_t *secret_key  /* OUT */,
                                                   const uint8_t *seed  /* IN */) {
   ml_kem_params params;
+  int res;
   ml_kem_512_params_init(&params);
-  return ml_kem_keypair_derand_ref(&params, public_key, secret_key, seed);
+  res = ml_kem_keypair_derand_ref(&params, public_key, secret_key, seed);
+#if defined(AWSLC_FIPS)
+  /* PCT failure is the only failure condition for key generation. */
+  if (res != 0) {
+      AWS_LC_FIPS_failure("ML-KEM keygen PCT failed");
+  }
+#else
+  assert(res == 0);
```

Comment:
Why capture here? All other code-paths normally throw it up the stack.

Reply:
This keeps the previous behavior from https://github.com/aws/aws-lc/pull/2195/files#diff-af6c334a9ca9f10f18987e623c41f74351fd662f0e01019d54a0716cdb710904L86. I'm OK either way.

Reply:
IMO remove

Reply:
Ack, will remove.

Reply:
Done.